llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -7276,6 +7276,12 @@ void SelectionDAGBuilder::visitIntrinsicCall(const CallInst &I,
     setValue(&I, getValue(I.getArgOperand(0)));
     return;
 
+  case Intrinsic::experimental_hot:
+    // Default lowering to false. It's intended to be lowered as soon as profile
+    // is avalible to unblock other optimizations.
```

Comment:
Missing globalisel and LowerIntrinsics handling

Reply:
we shouldn't even get to isel, this should be unconditionally lowered in the optimization pipeline and any instance of this intrinsic getting to isel should be considered a bug (e.g. coroutines, LowerMatrixIntrinsicsPass)

Reply:
The optimization pipeline isn't mandatory, -O0 must still work

Reply:
you are still required to run the -O0 "optimization" pipeline to do things like lowering (e.g. the examples I gave above) before codegen

Reply:
I guess we sill want some default lowering, e.g. for `-opt-bisect-limit=` working when we stop pipeline before the PGO based pass.

Reply:
moving out lowering into a separate CL